File: C++/tests/add_binary_test.cpp
#include <add_binary.hpp>
#include <gtest/gtest.h>
using namespace std;

class AddBinaryTest : public ::testing::Test {
protected:
  Solution solution;
};

TEST_F(AddBinaryTest, case1) {
  string a = "11";
  string b = "1";
  string want = "100";

  string actual = solution.addBinary(a, b);
  EXPECT_EQ(actual, want);
}

TEST_F(AddBinaryTest, case2) {
  string a = "1010";
  string b = "1011";
  string want = "10101";

  string actual = solution.addBinary(a, b);
  EXPECT_EQ(actual, want);
}

File: JavaScript/tests/add_binary.test.js
import { expect, test } from "vitest";
import { addBinary } from "#src/add_binary";

const cases = [
  { a: "11", b: "1", want: "100" },
  { a: "1010", b: "1011", want: "10101" },
  { a: "0", b: "0", want: "0" },
];

cases.forEach(({ a, b, want }, i) => {
  test(`case ${i + 1}`, () => {
    const actual = addBinary(a, b);
    expect(actual).toEqual(want);
  });
});
